optuna/integration/chainermn.py

```diff
@@ -190,180 +180,139 @@ def func() -> float:
-        # type: (str, Any) -> None
+    def set_system_attr(self, key: str, value: Any) -> None:
 
         if self.comm.rank == 0:
             assert self.delegate is not None
             self.delegate.set_system_attr(key, value)
         self.comm.mpi_comm.barrier()
 
     @property
-    def number(self):
-        # type: () -> int
-
-        def func():
-            # type: () -> int
+    def number(self) -> int:
+        def func() -> int:
 
             assert self.delegate is not None
             return self.delegate.number
 
         return self._call_with_mpi(func)
 
     @property
-    def _trial_id(self):
-        # type: () -> int
-
-        def func():
-            # type: () -> int
+    def _trial_id(self) -> int:
+        def func() -> int:
 
             assert self.delegate is not None
             return self.delegate._trial_id
 
         return self._call_with_mpi(func)
 
     @property
-    def params(self):
-        # type: () -> Dict[str, Any]
-
-        def func():
-            # type: () -> Dict[str, Any]
+    def params(self) -> Dict[str, Any]:
+        def func() -> Dict[str, Any]:
 
             assert self.delegate is not None
             return self.delegate.params
 
         return self._call_with_mpi(func)
 
     @property
-    def distributions(self):
-        # type: () -> Dict[str, BaseDistribution]
-
-        def func():
-            # type: () -> Dict[str, BaseDistribution]
+    def distributions(self) -> "Dict[str, BaseDistribution]":
+        def func() -> "Dict[str, BaseDistribution]":
 
             assert self.delegate is not None
             return self.delegate.distributions
 
         return self._call_with_mpi(func)
 
     @property
-    def user_attrs(self):
-        # type: () -> Dict[str, Any]
-
-        def func():
-            # type: () -> Dict[str, Any]
+    def user_attrs(self) -> Dict[str, Any]:
+        def func() -> Dict[str, Any]:
 
             assert self.delegate is not None
             return self.delegate.user_attrs
 
         return self._call_with_mpi(func)
 
     @property
-    def system_attrs(self):
-        # type: () -> Dict[str, Any]
-
-        def func():
-            # type: () -> Dict[str, Any]
+    def system_attrs(self) -> Dict[str, Any]:
+        def func() -> Dict[str, Any]:
 
             assert self.delegate is not None
             return self.delegate.system_attrs
 
         return self._call_with_mpi(func)
 
     @property
-    def datetime_start(self):
-        # type: () -> Optional[datetime]
-
-        def func():
-            # type: () -> Optional[datetime]
+    def datetime_start(self) -> Optional[datetime]:
+        def func() -> Optional[datetime]:
 
             assert self.delegate is not None
             return self.delegate.datetime_start
 
         return self._call_with_mpi(func)
 
-    def _call_with_mpi(self, func):
-        # type: (Callable) -> Any
+    def _call_with_mpi(self, func: "(Callable)") -> Any:
```

Comment:
Can we use `Callable` type instead of the string here?
```suggestion
    def _call_with_mpi(self, func: Callable) -> Any:
```